ui-v2/src/components/work-pools/work-pool-queue-create-dialog/use-create-work-pool-queue-form.ts

```diff
@@ -60,52 +65,119 @@ export const DEFAULT_VALUES = {
+}: UseCreateOrEditWorkPoolQueueFormOptions) => {
+	const { mutate: createWorkPoolQueue, isPending: isCreateLoading } =
 		useCreateWorkPoolQueueMutation();
+	const { mutate: updateWorkPoolQueue, isPending: isUpdateLoading } =
+		useUpdateWorkPoolQueueMutation();
 
 	const form = useForm({
 		resolver: zodResolver(formSchema),
 		defaultValues: DEFAULT_VALUES,
 	});
 
-	const create = (values: z.infer<typeof formSchema>) => {
+	// Sync form data with queue-to-edit data
+	useEffect(() => {
+		if (queueToEdit) {
+			form.reset({
+				name: queueToEdit.name,
+				description: queueToEdit.description ?? "",
+				is_paused: queueToEdit.is_paused ?? false,
+				concurrency_limit: queueToEdit.concurrency_limit ?? null,
+				priority: queueToEdit.priority ?? null,
+			});
+		} else {
+			form.reset(DEFAULT_VALUES);
+		}
+	}, [form, queueToEdit]);
+
+	const saveOrUpdate = (values: z.infer<typeof formSchema>) => {
 		const workQueueData = {
 			...values,
 			description: values.description || null,
 		};
 
-		createWorkPoolQueue(
-			{
-				workPoolName,
-				workQueueData,
-			},
-			{
-				onSuccess: () => {
-					toast.success("Work queue created");
-					form.reset(DEFAULT_VALUES);
-					onSubmit();
+		if (queueToEdit) {
+			updateWorkPoolQueue(
+				{
+					workPoolName,
+					queueName: queueToEdit.name,
+					workQueueData: {
+						...workQueueData,
+						is_paused:
+							workQueueData.is_paused ?? queueToEdit.is_paused ?? false,
+					},
+				},
+				{
+					onSuccess: () => {
+						toast.success("Work queue updated");
+						form.reset(DEFAULT_VALUES);
+						onSubmit();
+					},
+					onError: (error) => {
+						const message =
+							error.message || "Unknown error while updating work queue.";
+						form.setError("root", { message });
+					},
+				},
+			);
+		} else {
+			createWorkPoolQueue(
+				{
+					workPoolName,
+					workQueueData,
 				},
-				onError: (error) => {
-					const message =
-						error.message || "Unknown error while creating work queue.";
-					form.setError("root", { message });
+				{
+					onSuccess: () => {
+						toast.success("Work queue created");
+						form.reset(DEFAULT_VALUES);
+						onSubmit();
+					},
+					onError: (error) => {
+						const message =
+							error.message || "Unknown error while creating work queue.";
+						form.setError("root", { message });
+					},
 				},
-			},
-		);
+			);
+		}
 	};
 
 	return {
 		form,
-		create,
+		saveOrUpdate,
+		isLoading: isCreateLoading || isUpdateLoading,
+	};
+};
+
+// Backwards compatible alias for create-only usage
+type UseCreateWorkPoolQueueFormOptions = {
+	workPoolName: string;
+	onSubmit: () => void;
+};
+
+export const useCreateWorkPoolQueueForm = ({
+	workPoolName,
+	onSubmit,
+}: UseCreateWorkPoolQueueFormOptions) => {
+	const { form, saveOrUpdate, isLoading } = useCreateOrEditWorkPoolQueueForm({
+		workPoolName,
+		onSubmit,
+	});
+
+	return {
+		form,
+		create: saveOrUpdate,
 		isLoading,
 	};
 };
```

Comment:
You don't need to worry about backwards compatibiliy. Feel free to delete the old implementation as long as you update all the places where the old implementation was used to use the new implementation.